sdcm/utils/replication_strategy_utils.py

```diff
@@ -95,3 +95,23 @@ def __call__(self, **keyspaces: ReplicationStrategy) -> None:
         for keyspace, strategy in keyspaces.items():
             self._preserve_replication_strategy(keyspace)
             strategy.apply(self.node, keyspace)
+
+
+class KeyspaceReplicationStrategyInfo(NamedTuple):
+    keyspace_name: str
+    replication_factor: list[int]
+    replication_strategy: ReplicationStrategy
+
+
+def set_keyspace_info(keyspace_name: str, db_node: BaseNode):
```

Comment:
this is getting keyspace info - not setting. Use `get` in the name.
Also `keyspace_info` is very broad, while we take only rf and strategy.

My proposal is to enhance each `ReplicationStrategy` subclasses to include `replication_factors` which will be a list of factors like calculated here. Then also `KeyspaceReplicationStrategyInfo` will become redundant.

Reply:
changed accordingly to comments ( I hope)

Reply:
yes, direction is right. There are few more comments only.